src/exporters/USDZExporter.ts

```diff
@@ -0,0 +1,457 @@
+    const name = 'Geometry'
+    const attributes = geometry.attributes
+    const count = attributes.position.count
+
+    return `
+    def Mesh "${name}"
+    {
+        int[] faceVertexCounts = [${this.buildMeshVertexCount(geometry)}]
+        int[] faceVertexIndices = [${this.buildMeshVertexIndices(geometry)}]
+        normal3f[] normals = [${this.buildVector3Array(attributes.normal, count)}] (
+            interpolation = "vertex"
+        )
+        point3f[] points = [${this.buildVector3Array(attributes.position, count)}]
+        float2[] primvars:st = [${this.buildVector2Array(attributes.uv, count)}] (
+            interpolation = "vertex"
+        )
+        uniform token subdivisionScheme = "none"
+    }
+`
+  }
+
+  private buildMeshVertexCount(geometry: BufferGeometry): string {
+    const count = geometry.index !== null ? geometry.index.array.length : geometry.attributes.position.count
+
+    return Array(count / 3)
+      .fill(3)
+      .join(', ')
+  }
+
+  private buildMeshVertexIndices(geometry: BufferGeometry): string {
+    if (geometry.index !== null) {
+      // @ts-expect-error
+      return geometry.index.array.join(', ')
+    }
+
+    const array = []
+    const length = geometry.attributes.position.count
+
+    for (let i = 0; i < length; i++) {
+      array.push(i)
+    }
+
+    return array.join(', ')
+  }
+
+  private buildVector3Array(attribute: BufferGeometry['attributes'][number], count: number): string {
+    if (attribute === undefined) {
+      console.warn('USDZExporter: Normals missing.')
+      return Array(count).fill('(0, 0, 0)').join(', ')
+    }
+
+    const array = []
+    const data = attribute.array
+
+    for (let i = 0; i < data.length; i += 3) {
+      array.push(
+        `(${data[i + 0].toPrecision(this.PRECISION)}, ${data[i + 1].toPrecision(this.PRECISION)}, ${data[
+          i + 2
+        ].toPrecision(this.PRECISION)})`,
+      )
+    }
+
+    return array.join(', ')
+  }
+
+  private buildVector2Array(attribute: BufferGeometry['attributes'][number], count: number): string {
+    if (attribute === undefined) {
+      console.warn('USDZExporter: UVs missing.')
+      return Array(count).fill('(0, 0)').join(', ')
+    }
+
+    const array = []
+    const data = attribute.array
+
+    for (let i = 0; i < data.length; i += 2) {
+      // @ts-expect-error
+      array.push(`(${data[i + 0].toPrecision(this.PRECISION)}, ${1 - data[i + 1].toPrecision(this.PRECISION)})`)
+    }
+
+    return array.join(', ')
+  }
+
+  // Materials
+  private buildMaterials(materials: USDZExporter['materials'], textures: USDZExporter['textures']): string {
+    const array = []
+
+    for (const uuid in materials) {
+      const material = materials[uuid]
+
+      array.push(this.buildMaterial(material, textures))
+    }
+
+    return `def "Materials"
+{
+${array.join('')}
+}
+`
+  }
+
+  private buildMaterial(material: MaterialRepresentaion, textures: USDZExporter['textures']): string {
```

Comment:
looks like the second arg isn't used, could we remove it?

Reply:
Yes, it seem to be unused. I'll remove it.